core/core-impl/src/main/java/io/fabric8/launcher/core/impl/documentation/BoosterDocumentationStoreImpl.java

```diff
@@ -0,0 +1,127 @@
+
+    private static final String DOCUMENTATION_REPOSITORY = "https://github.com/fabric8-launcher/launcher-documentation.git";
+
+    private static final String DOCUMENTATION_BRANCH = "master";
+
+    private volatile CompletableFuture<Path> pathCompletableFuture;
+
+    private final ExecutorService async;
+
+    private DocumentationFilesDownloader downloader;
+
+    @Inject
+    public BoosterDocumentationStoreImpl(final ManagedExecutorService managedExecutorService) {
+        this(managedExecutorService, () -> cloneGitRepository(DOCUMENTATION_REPOSITORY, DOCUMENTATION_BRANCH));
+    }
+
+    //Visible for testing
+    BoosterDocumentationStoreImpl(final ExecutorService async) {
+        this(async, () -> cloneGitRepository(DOCUMENTATION_REPOSITORY, DOCUMENTATION_BRANCH));
+    }
+
+    //Visible for testing
+    BoosterDocumentationStoreImpl(final ExecutorService async, final DocumentationFilesDownloader downloader) {
+        this.async = requireNonNull(async, "async must be specified");
+        this.downloader = requireNonNull(downloader, "downloader must be specified");
+    }
+
+    // Initialize on startup
+    public void init(@Observes @Initialized(ApplicationScoped.class) Object init) {
+        // Do nothing
+    }
+
+    @PostConstruct
+    public void initialize() {
+        getDocumentationPath();
+    }
+
+    @Override
+    public CompletableFuture<Path> reloadDocumentation() {
+        return getDocumentationPath(true);
+    }
+
+    @Override
+    public synchronized CompletableFuture<Path> getDocumentationPath() {
+        return getDocumentationPath(false);
+    }
+
+    @Override
+    public void waitForDocumentation() throws ExecutionException, InterruptedException {
+        getDocumentationPath().get();
+    }
+
+    private synchronized CompletableFuture<Path> getDocumentationPath(final boolean reload) {
+        if (reload || pathCompletableFuture == null) {
+            pathCompletableFuture = createDocumentationPathFuture();
+        }
+        return pathCompletableFuture;
+    }
+
+    private CompletableFuture<Path> createDocumentationPathFuture() {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return downloader.download();
+            } catch (final IOException e) {
+                logger.log(Level.SEVERE, "Error while loading documentation", e);
+                throw new IllegalStateException("Error while loading documentation", e);
+            }
+        }, async);
+    }
+
+    private static Path cloneGitRepository(final String readmeRepositoryURI, final String branch) throws IOException {
+        logger.log(Level.INFO, "Indexing contents from {0} using {1} ref",
+                   new Object[]{readmeRepositoryURI, branch});
+        final Path catalogPath = Files.createTempDirectory("booster-readme");
+        logger.info("Created " + catalogPath);
+        ProcessBuilder builder = new ProcessBuilder()
```

Comment:
Any reason to not use Git sparse-checkout here? https://briancoyner.github.io/2013/06/05/git-sparse-checkout.html

Reply:
We talked about it with @quintesse, and though it could wait:
- it was easier to code this way in a first shot (1 command instead of multiple)
- it may change soon depending on the documentation future location